addon/components/x-select.js

```diff
@@ -200,6 +200,10 @@ export default Ember.Component.extend({
    */
   unregisterOption: function(option) {
     this.get('options').removeObject(option);
-    this._updateValueSingle();
+
+    // We don't want to update the value if we're tearing the component down.
+    if (!this.get('isDestroying')) {
+      this._updateValueSingle();
```

Comment:
Also should this check if `multiple=true` is set and do `this._updateValueMultiple()`?

```js
if (!this.get('isDestroying')) {
  if (this.get('multiple')) {
    this._updateValueMultiple();
  } else {
    this.updateValueSingle();
  }
} 
```